proposals/theme_repo.md

```diff
@@ -0,0 +1,102 @@
+# Theme repo standard proposal
+
+## Description Of Issue
+
+Theme repositories are not standardised, outside of any de facto standards from theme loaders
+
+## Solution
+
+Standardise how repos specify the theme list, and how they expose their own metadata
+
+## Advantages
+
+- repos can provide metadata such as their name, maintainer, etc
+- all repos work the same way
+
+## Disadvantages
+
+- other theme loaders may wish to add additional features, which this would lock down
+
+## General Implementation Details
+
+### Manifest
+
+I propose the following JSON Schema:
+
+```json
+{
+  "$schema": "https://json-schema.org/draft/2020-12/schema",
+  "$id": "N/A",
+  "title": "Cumcord Theme Repo Manifest",
+  "description": "The manifest format for Cumcord theme repos",
+  "type": "object",
+  "properties": {
+    "meta": {
+      "description": "Repo metadata",
+      "type": "object",
+      "properties": {
+        "name": {
+          "description": "The name of the repo",
+          "type": "string"
+        },
+        "description": {
+          "description": "A more detailed description of the repo",
+          "type": "string"
+        },
+        "maintainer": {
+          "description": "Who maintains the repo",
+          "type": "string"
+        }
+      },
+      "required": [
+        "name",
+        "description",
+        "maintainer"
+      ]
+    },
+    "themes": {
+      "description": "The list of themes in the repo, either relative or absolute URLs",
+      "type": "array",
+      "items": {
+        "type": "string"
+      }
+    }
+  },
+  "required": [
+    "meta",
+    "themes"
+  ]
+}
+```
+
+which validates correctly against the following example repo, which demonstrates metadata, relative urls, and absolute themes:
+
+```json
+{
+  "meta": {
+    "name": "My awesome repo",
+    "maintainer": "generic author",
+    "description": "A place for awesome themes to go"
+  },
+  "themes": [
+    "https://raw.githubusercontent.com/generic_author/generic_theme/master/generic_theme.theme.css",
+    "test_1/test_theme.css",
+  ]
```

Comment:
@Cr3atable What do you think about this?

Reply:
How does this handle theme manifests? @yellowsink

Reply:
As discussed in #2 but not included in the proposal, `https://example.com/theme.css/../cumcord_theme.json` - go up one and then down to the json